kratos/processes/apply_double_table_process.hpp

```diff
@@ -0,0 +1,147 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ignasi de Pouplana
+//  Collaborator:    Alejandro Cornejo
+//
+
+#if !defined(KRATOS_APPLY_DOUBLE_TABLE_PROCESS)
+#define  KRATOS_APPLY_DOUBLE_TABLE_PROCESS
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "apply_component_table_process.hpp"
+// #include "fem_to_dem_application_variables.h"
+
+namespace Kratos
+{
+
+/// The base class for all processes in Kratos.
+/** This function applies a value according to a table defined in the mdpa
+*/
+class ApplyDoubleTableProcess : public ApplyComponentTableProcess
+{
+    
+public:
+
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of ApplyConstantScalarValueProcess
+    KRATOS_CLASS_POINTER_DEFINITION(ApplyDoubleTableProcess);
+
+    /// Constructor
+    ApplyDoubleTableProcess(ModelPart& model_part, Parameters rParameters) 
+        :ApplyComponentTableProcess(model_part, rParameters) {}
+
+    
+    /// Destructor
+    ~ApplyDoubleTableProcess() override {}
+
+    /// Execute method is used to execute the ApplyDoubleTableProcess algorithms.
+    void Execute() override
+    {
+    }
+    
+    /// this function is designed for being called at the beginning of the computations
+    /// right after reading the model and the groups
+    void ExecuteInitialize() override
+    {
+        KRATOS_TRY;
+        
+        Variable<double> var = KratosComponents< Variable<double> >::Get(mvariable_name);
+        
+        const int nnodes = static_cast<int>(mr_model_part.Nodes().size());
+
+        if (nnodes != 0) {
+            ModelPart::NodesContainerType::iterator it_begin = mr_model_part.NodesBegin();
+
+            #pragma omp parallel for
+            for (int i = 0; i < nnodes; i++) {
+                ModelPart::NodesContainerType::iterator it = it_begin + i;
+                if (mis_fixed) {
+                    it->Fix(var);
+                }
+                it->FastGetSolutionStepValue(var) = minitial_value;
+            }
+        }
+        KRATOS_CATCH("");
+    }
+
+    /// this function will be executed at every time step BEFORE performing the solve phase
+    void ExecuteInitializeSolutionStep() override
+    {
+        KRATOS_TRY;
+        
+        Variable<double> var = KratosComponents< Variable<double> >::Get(mvariable_name);
+        
+        const double time = mr_model_part.GetProcessInfo()[TIME];
```

Comment:
can you make this variable?
I think it would increase the useability a lot if you wouldn't hardcode to time